src/components/PoolingWidget/DefineSpread.tsx

```diff
@@ -1,52 +1,48 @@
-import React from 'react'
-import { TokenDetails } from '@gnosis.pm/dex-js'
+import React, { useCallback } from 'react'
 
-import { SpreadInformationWrapper, DefineSpreadWrapper } from './DefineSpread.styled'
+import { DefineSpreadWrapper } from './DefineSpread.styled'
 
-import { DEFAULT_DECIMALS } from 'const'
+import InputWithTooltip from '../InputWithTooltip'
 
-interface DefineSpreadProps {
-  selectedTokensMap: Map<number, TokenDetails>
+import { useFormContext, FieldError } from 'react-hook-form'
 
-  spread: number
-  setSpread: React.Dispatch<React.SetStateAction<number>>
+function formatErrorMessage(errorString: string): string {
+  const cleanedString = errorString.replace(/"/g, '')
+  return cleanedString[0].toUpperCase() + cleanedString.slice(1)
 }
 
-type SpreadInformationProps = Omit<DefineSpreadProps, 'setSpread'>
-
-export const SpreadInformation: React.FC<SpreadInformationProps> = ({ selectedTokensMap, spread }) => {
-  const tokenSymbolsString = React.useMemo(() => Array.from(selectedTokensMap.values()).map(token => token.symbol), [
-    selectedTokensMap,
-  ])
-
-  return (
-    <SpreadInformationWrapper>
-      <strong>Sell Spread</strong>
-      <p>
-        {tokenSymbolsString.join(', ')} for <b>at least</b> <br />
-        <i>${(1 + spread / 100).toFixed(DEFAULT_DECIMALS)}</i>
-      </p>
-      <strong>Buy Spread</strong>
-      <p>
-        {tokenSymbolsString.join(', ')} for <b>at most</b> <br />
-        <i>${(1 - spread / 100).toFixed(DEFAULT_DECIMALS)}</i>
-      </p>
-    </SpreadInformationWrapper>
-  )
+interface DefineSpreadProps {
+  isSubmitting: boolean
+  spread: number
 }
 
-const DefineSpread: React.FC<DefineSpreadProps> = ({ selectedTokensMap, spread, setSpread }) => {
-  const handleSpreadChange = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>): void => {
-    if (+value < 0 || +value >= 100) return
+const DefineSpread: React.FC<DefineSpreadProps> = ({ isSubmitting }) => {
+  const { errors, register } = useFormContext()
 
-    setSpread(+value)
-  }
+  const onKeyPress = useCallback((event: React.KeyboardEvent<HTMLInputElement>): void => {
+    if (event.key === 'Enter') {
+      event.preventDefault()
+    }
+  }, [])
+
+  const errorMessage = (errors?.spread as FieldError)?.message
 
   return (
     <DefineSpreadWrapper>
-      <strong>Spread %</strong>
-      <input type="number" step="0.1" value={spread} onChange={handleSpreadChange} />
-      <SpreadInformation selectedTokensMap={selectedTokensMap} spread={spread} />
+      <p>
+        <strong>Spread %</strong> -{' '}
+        <small>percentage you want to sell above $1, and buy below $1 between all selected tokens</small>
+      </p>
+      <InputWithTooltip
+        name="spread"
+        type="number"
+        step={0.1}
+        disabled={isSubmitting}
+        ref={register}
+        onKeyPress={onKeyPress}
+        tooltip={(errorMessage && formatErrorMessage(errorMessage)) || 'Value between 0 and 100, not inclusive'}
```

Comment:
I'm not sure it's a good idea to use tooltips to report form errors. Not without styling tooltips in red colors or something, at least

Reply:
done, good advice